checks.d/teamcity.py

```diff
@@ -0,0 +1,126 @@
+
+            last_build_id = resp.json().get('build')[0].get('id')
+        except requests.exceptions.HTTPError:
+            if resp.status_code == 401:
+                self.log.error("Access denied. You must enable guest authentication")
+            self.log.error(
+                "Failed to retrieve last build ID with code {0} for instance '{1}'"
+                .format(resp.status_code, instance_name)
+            )
+            raise
+        except Exception:
+            self.log.exception(
+                "Unhandled exception to get last build ID for instance '{0}'"
+                .format(instance_name)
+            )
+            raise
+
+        self.log.debug(
+            "Last build id for instance {0} is {1}."
+            .format(instance_name, last_build_id)
+        )
+        self.last_build_ids[instance_name] = last_build_id
+
+    def _build_and_send_event(self, new_build, instance_name, is_deployment, host, tags):
+        self.log.info("Found new build with id {0}, saving and alerting.".format(new_build["id"]))
+        self.last_build_ids[instance_name] = new_build["id"]
+
+        event_dict = {
+            'timestamp': int(time.time()),
+            'source_type_name': 'teamcity',
+            'host': host,
+            'tags': [],
+        }
+        if is_deployment:
+            event_dict['event_type'] = 'teamcity_deployment'
+            event_dict['msg_title'] = "{0} deployed to {1}".format(instance_name, host)
+            event_dict['msg_text'] = "Build Number: {0}\n\nMore Info: {1}"\
+                                        .format(new_build["number"], new_build["webUrl"])
+            event_dict['tags'].append('deployment')
+        else:
+            event_dict['event_type'] = "build"
+            event_dict['msg_title'] = "Build for {0} successful".format(instance_name)
+
+            event_dict['msg_text'] = "Build Number: {0}\nDeployed To: {1}\n\nMore Info: {2}"\
+                                        .format(new_build["number"], host, new_build["webUrl"])
+            event_dict['tags'].append('build')
+
+        if tags:
+            event_dict["tags"].extend(tags)
+
+        self.event(event_dict)
+
+    def check(self, instance):
+        instance_name = instance.get('name')
+        if instance_name is None:
+            raise Exception("Each instance must have a unique name")
+
+        build_configuration = instance.get('build_configuration')
+        if build_configuration is None:
+            raise Exception("Each instance must have a build configuration")
+
+        host = instance.get('host_affected') or self.hostname
+        tags = instance.get('tags')
+        is_deployment = _is_affirmative(instance.get('is_deployment', False))
+
+        self._initialize_if_required(instance_name, build_configuration)
+
+        # Look for new successful builds
+        new_build_url = "http://{0}/guestAuth/app/rest/builds/?locator=buildType:{1},sinceBuild:id:{2},status:SUCCESS"\
```

Comment:
it's a nitpick but let's declare this string as a constant
```python
BUILD_URL = "http://{0}/guestAuth/app/rest/builds/?locator=buildType:{1},sinceBuild:id:{2},status:SUCCESS"
```
and then you can do:
```python
new_build_url = BUILD_URL.format(self.server, build_configuration, self.last_build_ids[instance_name])
```